proto/viam/app/mltraining/v1/ml_training.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package viam.app.mltraining.v1;
+
+import "app/data/v1/data.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/app/mltraining/v1";
+
+service MLTrainingService {
+  rpc SubmitTrainingJob(SubmitTrainingJobRequest) returns (SubmitTrainingJobResponse);
+  rpc GetTrainingJob(GetTrainingJobRequest) returns (GetTrainingJobResponse);
+}
+
+enum ModelType {
+  MODEL_TYPE_UNSPECIFIED = 0;
+  MODEL_TYPE_MULTI_CLASS_CLASSIFICATION = 1;
+  MODEL_TYPE_MULTI_LABEL_CLASSIFICATION = 2;
+}
+
+message SubmitTrainingJobRequest {
+  viam.app.data.v1.Filter filter = 1;
+  string org_id = 2;
+  string model_name = 3;
+  string model_version = 4;
+  ModelType model_type = 5;
+  repeated string training_labels = 6;
```

Comment:
Since this can be expanded later to non-string labels, I think should remain consistent with our API and switch to `repeated string tags` 

The other option is to make this a nested oneof Label type, but decided against that in our existing API for 1) simplicity and 2) since tags will likely be the most common label type.

Reply:
Updated to tags. 

Is the idea that in the future we'd deprecate this tags field in favor of some more general label field? Or would it be in addition to it?

Reply:
Thinking through. In the Data API it makes sense to keep separate as there could be different types of labels for different purposes on a single datum, and the user will usually want to grab all tags _or_ bounding boxes _or_ segmentation masks for each distinct model type, whereas here we'd likely pass one type in.

We've also been careful about using oneofs which are a bit more clunky to set/get/understand — vs just listing all the fields — but otherwise I'd see this as equally valid:

```
message Labels {
   oneof {
        repeated string tags = 1; 
   }
}
```

Reply:
I'm gonna ask Eric about the oneof thing, I've never really gotten why we avoid them as a general rule.

In that case though it sounds like we'd start with just tags, but presumably deprecate that field in the medium term future in favor of some `Labels` once we add the ability to train models on different label types (+ add the ability to add those labels)?

Reply:
SGTM!